src/measure_theory/set_integral.lean

```diff
@@ -104,58 +99,62 @@ begin
   exact (hs.inter_preimage hB hsm).union (ht.inter_preimage hB htm)
 end
 
-end has_zero
+end set
 
 section integrable_on
 variables [measure_space α] [normed_group β] {s t : set α} {f g : α → β}
 
 /-- `integrable_on s f` means `f` is integrable over the set `s`. -/
-@[reducible]
-def integrable_on (s : set α) (f : α → β) : Prop := integrable (indicator s f)
+def set.integrable_on (s : set α) (f : α → β) : Prop := integrable (s.indicator f)
+
+open set
 
 lemma integrable_on_congr (h : ∀x, x ∈ s → f x = g x) : integrable_on s f ↔ integrable_on s g :=
 by simp only [integrable_on, indicator_congr h]
 
-lemma integrable_on_congr_ae (h : ∀ₘx, x ∈ s → f x = g x) :
+lemma integrable_on_congr_ae (h : ∀ₘ x, x ∈ s → f x = g x) :
   integrable_on s f ↔ integrable_on s g :=
 by { apply integrable_congr_ae, exact indicator_congr_ae h }
 
 @[simp] lemma integrable_on_empty (f : α → β) : integrable_on ∅ f :=
 by { simp only [integrable_on, indicator_empty], apply integrable_zero }
 
-lemma integrable_on.subset (h : s ⊆ t) : integrable_on t f → integrable_on s f :=
+lemma integrable.integrable_on (s : set α) (hf : integrable f) : integrable_on s f :=
+by { refine integrable_of_le (λa, _) hf, apply norm_indicator_le_norm_self }
+
+lemma set.integrable_on.subset (h : s ⊆ t) : integrable_on t f → integrable_on s f :=
 by { apply integrable_of_le_ae, filter_upwards [] norm_indicator_le_of_subset h _ }
 
 variables {𝕜 : Type*} [normed_field 𝕜] [normed_space 𝕜 β]
 
-lemma integrable_on.smul (s : set α) (c : 𝕜) {f : α → β} :
+lemma set.integrable_on.smul (s : set α) (c : 𝕜) {f : α → β} :
   integrable_on s f → integrable_on s (λa, c • f a) :=
 by { simp only [integrable_on, indicator_smul], apply integrable.smul }
 
-lemma integrable_on.mul_left (s : set α) (r : ℝ) {f : α → ℝ} (hf : integrable_on s f) :
+lemma set.integrable_on.mul_left (s : set α) (r : ℝ) {f : α → ℝ} (hf : integrable_on s f) :
```

Comment:
I am really sorry about this, but I think my previous suggestion (to use the `set` namespace here) is really bad. This is just like `continuous_on` or `differentiable_on`: you want to be able to write `hf.mul_left` when `hf : integrable_on s f`, so this should not be in the set namespace. But then I think it should be the same for `measurable_on`, i.e., it should not be in the set namespace either. Sorry again.

Reply:
Oh, sure. No problem.